### PR TITLE
[WIP]: Add invidious.js connector

### PR DESCRIPTION
**Pull request by doronbehar**
Nothing really works yet, but instead of discussing it over at #13, it might be more comfortable here.

### CODE REVIEW

**Comment by f1u77y** [comment]
Answering the questing from previous issue:

> One more question please regarding the mediaSelector: I tried to set it to just video with the hope that it will eventually make the playPause() functions work just like in YouTube but it didn't work.

`playPause()` uses `playButtonSelector`, so you might need set it for this function to work.

And I think you should add the js file name to the `directoryContents` variable just like other connectors. I know it's awful but I'll fix it later.

**Comment by doronbehar** [comment]
> And I think you should add the js file name to the directoryContents variable just like other connectors. I know it's awful but I'll fix it later.

LOL I've managed to figure that out right before I read your comment :smile_cat:. That's available in https://github.com/f1u77y/web-media-controller/pull/14/commits/45c3d05b7d8981a8592c49442f8cf890834acccf

Actually, I think the PR is totally ready now. I've used both `mediaSelector` and `playButtonSelector` eventually.

File: connectors/invidious.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import BaseConnector from 'content/base-connector';
+import Utils from 'content/utils';
+import _ from 'underscore';
+
+new class extends BaseConnector {
+    constructor() {
+        super();
+        this.name = 'Invidious';
+        this.prefix = '/com/invidious';
```
Comment (f1u77y) [inline connectors/invidious.js]: This should be hostname in reverse order, eg. `/us/invido`. Looks ugly but I think we should be consistent in prefix scheme.

File: connectors/invidious.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    get uniqueId() {
+        const params = new URLSearchParams(location.search.substr(1));
+        return Promise.resolve(params.get('v')
+                               .replace('_', '_u')
+                               .replace('-', '_d'));
+    }
+
+    get controlsInfo() {
+        const hasPrevButton = document.querySelector(this.prevButtonSelector) !== null;
```
Comment (f1u77y) [inline connectors/invidious.js]: There is no `prevButtonSelector` in your connector. Did you forget to remove this line?

File: connectors/invidious.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        return super.controlsInfo.then(controlsInfo => _(controlsInfo).extend({
+            canStop: false,
+            canGoPrevious: hasPrevButton,
```
Comment (f1u77y) [inline connectors/invidious.js]: I think `canGoPrevious` and `canGoNext` are always false unless this website has some kind of playlists.

**Review by f1u77y — CHANGES_REQUESTED** [review]

**Comment by f1u77y** [comment]
Seems fine now. Squash your commits please, and then I will merge.

**Review by f1u77y — APPROVED** [review]

**Comment by doronbehar** [comment]
Thanks for the merge. Could you please add a new release? I thought I'd be able to build it my self but I get errors similar to the ones I reported in #13:

```
Loading "Gruntfile.js" tasks...ERROR
>> Error: Unable to read "cws.json" file (Error code: ENOENT).

Running "build:firefox" (build) task

Running "clean:common" (clean) task
Verifying property clean.common exists in config...ERROR
>> Unable to process task.
Warning: Required config property "clean.common" missing. Use --force to continue.

Aborted due to warnings.
```

**Pull request merged**